openapi-cli/src/test/java/io/ballerina/openapi/generators/schema/NegativeConstraintTests.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com). All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.openapi.generators.schema;
+
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.openapi.core.GeneratorUtils;
+import io.ballerina.openapi.core.exception.BallerinaOpenApiException;
+import io.ballerina.openapi.core.generators.schema.BallerinaTypesGenerator;
+import io.ballerina.openapi.generators.common.TestUtils;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+import io.swagger.v3.oas.models.OpenAPI;
+import org.ballerinalang.formatter.core.FormatterException;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import static io.ballerina.openapi.generators.common.TestUtils.getDiagnostics;
+import static org.testng.Assert.assertFalse;
+
+/**
+ * This test class contains negative tests related to constraint validation.
+ */
+public class NegativeConstraintTests {
+    private static final Path RES_DIR = Paths.get("src/test/resources/generators/schema").toAbsolutePath();
+
+    @Test(description = "Tests for non-string type record field has invalid pattern constraint." +
+            "There is no pattern validation within swagger parser although swagger support for only string value to " +
+            "have regular pattern" +
+            "(https://swagger.io/docs/specification/data-models/data-types/#string:~:text=The%20pattern%20keyword%20)" +
+            "Therefore here code generation for pattern with non-string type silently ignored. This will handle " +
+            "separate PR in future.")
```

Comment:
```suggestion
            "Therefore, code generation for patterns with non-string types is silently ignored here. This will be handled " +
            "in a separate PR in the future.")
```